### ISSUE TITLE
Roslyn unable to load analyzer configs if disk letter in path of current directory is not matching file name in analyzerconfig option

### ISSUE THREAD

**Issue by ghisguth** [dotnet/roslyn#50300]
**Version Used**: 

Microsoft (R) Visual C# Compiler version 3.8.0-5.20604.10 (9ed4b774)

**Steps to Reproduce**:

1. Open command line. Go to directory with .editorconfig. Be sure to change directory using lowercase for drive. E.g. `cd c:\Src\Foo\Bar\`
2. Create analyzer config in c:\Src\Foo\.editorconfig
2. Execute csc.exe with `/analyzerconfig:C:\Src\Foo\.editorconfig` (Note disk letter is upper case `C:`)

**Expected Behavior**:

Roslyn will load editor config and apply to all files in `c:\Src\Foo\Bar\`

**Actual Behavior**:

Roslyn ignores the config for analyzers.

** Details **:

* Roslyn loads base directory using `Directory.GetCurrentDirectory()`.
https://github.com/dotnet/roslyn/blob/26939574c8e180d5135b2a28f183709b67483634/src/Compilers/Shared/BuildClient.cs#L92
* All source files will have base path appended in ParseFileArgument:
https://github.com/dotnet/roslyn/blob/52b08d4f4c2895a2d39a0429d82317775b814754/src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs#L953
* AnalyzerConfig will load file as is and store NormalizedDirectory with the same casing it was passed.
https://github.com/dotnet/roslyn/blob/8e23563ca8ce49f2f61bea49178cae141605edae/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs#L150-L151
* AnalyzerConfigSet.GetOptionsForSourcePath will use StartsWith with StringComparison.Ordinal to check if c# file belongs to base directory of the config. So if casing of base directory and analyzerconfig option is not matching, it will not apply analyzer configs to it.
https://github.com/dotnet/roslyn/blob/8e23563ca8ce49f2f61bea49178cae141605edae/src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs#L213-L217

** Visual Studio impact**:

if you open csproj via solution. Visual studio will lowercase disk in path of the csproj and while compile all paths will start from lowercase (e.g. c:\Src\Foo\Bar\Bar.csproj). analyzerconfig option will work as expected.

If you open csproj directly and try to compile. Visual studio will open it with disk as upper case (e.g. C:\Src\Foo\Bar\Bar.csproj). And analyzer config will be ignored.

**Comment by ghisguth** [comment]
+ @jasonmalinowski FYI. Seems like either either string comparison should be case insensitive. Or normalized path should always lowercase disk letter if it is present.

**Comment by jasonmalinowski** [comment]
@ghisguth Thanks for the detailed analysis here; we'll have to process this one to figure out which layers will need a fix here!

**Comment by jaredpar** [comment]
I don't think this discussion is only limited to `/analyzerConfig`. Consider that for options like `/pathMap` we also treat the argument literally and do no normalization on it.

How we handle file path casing is an aspect of the compiler that I don't think we have an explicit documented plan on. This is likely a good time to do that and add a bit of consistency to how we do it. The general approach we've used to the best of my recollection is:

- Basic path normalization on arguments to the command line. That can include anything from removing trailing slashes on directories, getting consistent casing for drive letters, etc ... 
- All other parts of the compiler treat paths as ordinal ignore case. 

@chsienki @mavasani @tmat for feedback here as well.
